### PR TITLE
Add e2e test details to dev guide

### PR DESCRIPTION
**Pull request by hikarukin**
Add English language doc describing setup of e2e tests to dev guide.

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [X] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: docs/development-guide/e2e-test-details.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
+# E2E test cases for opr-paas
+
+Below you will find a list of features that we want to test using end-to-end testing.
+For each feature, the setup and the assessments are listed.
+
+TODO:
+1. Take up `RoleMappings` in `testConfig`;
```
Comment (portly-halicore-76) [inline docs/development-guide/e2e-test-details.md]: Already done in the meantime.

File: docs/development-guide/e2e-test-details.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+2. Test `extra_permissions` / `default_permissions` in config;
+
+## Paas
```
Comment (portly-halicore-76) [inline docs/development-guide/e2e-test-details.md:6]: Do we need to test this specifically as we test each 'feature' of Paas in detail as described below. What is the value of these tests?

File: docs/development-guide/e2e-test-details.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+   **Given** that a specified `PaasNs` exists,<br/>
+   **when** the `PaasNs` is deleted,<br/>
+   **then** the namespace belonging to the `PaasNs` is also removed.
+
+Post scenarios: reset environment to clean slate.
+
+## Namespaces
+
+1. A minimal `Paas` configuration without namespaces results in one namespace.<br/><br/>
+   **Give** a minimal `Paas` configuration without namespaces,<br/>
```
Comment (portly-halicore-76) [inline docs/development-guide/e2e-test-details.md]: Given*

File: docs/development-guide/e2e-test-details.md (hunk 2 above)
Comment (hikarukin) [reply docs/development-guide/e2e-test-details.md:6]: They were in the original text. Removed.

**Review by AxiomaticFixedChimpanzee — APPROVED** [review]

**Pull request merged**